### PR TITLE
Fix language server path displayed in LSP status tooltip

### LINKED ISSUE #52822: LSP status tooltip shows the `node` binary path instead of the language server's script path.
### Reproduction steps

1. Start Zed and open a project that uses an LSP managed by Zed/npm (e.g., `basedpyright` or `pyright`).
2. Ensure the language server is downloaded and managed by Zed.
3. Click the LSP icon in the status bar to open the language servers list.
4. Hover over the "running" status for the active server.
5. The tooltip displays the path to the node executable instead of the path to the language server script itself.

Here is a screenshot:

Image

### Current vs. Expected behavior

Current:
The tooltip displays the path to the node binary.

Expected:
The tooltip should display the actual path to the language server.

### Zed version and system specs

Zed: v0.230.0+preview.205.9437a84390a396d666f04b38db87d89bb07284c1 (Zed Preview) 
OS: Windows 10.0.26200
Memory: 47.8 GiB
Architecture: x86_64
GPU: Intel(R) UHD Graphics 770 || Intel Corporation || 32.0.101.7085

### Attach Zed log file

<details><summary>Zed.log</summary>

```log

```

</details>

### Relevant Zed settings

<details><summary>settings.json</summary>

```json

```

</details>

### Relevant Keymap

<details><summary>keymap.json</summary>

```json

```

</details>

### (for AI issues) Model provider details

_No response_

### If you are using WSL on Windows, what flavor of Linux are you using?

None

### PR DESCRIPTION
**Pull request by lingyaochu**
# Objective

Closes #52822

When hovering over the LSP status tooltip for a running language server, the binary path is displayed. Currently, this tooltip shows `LanguageServerBinary.path`. However, many language servers are executed through runtimes such as Node or Python. For example, Zed-managed `Basedpyright` produces a `LanguageServerBinary` like:
```Rust
LanguageServerBinary {
    path: "/usr/bin/node",
    arguments: [
        "/home/xin/.local/share/zed/languages/basedpyright/node_modules/basedpyright/langserver.index.js",
        "--stdio",
    ],
    env: ...
}
``` 
In this case, only `"/usr/bin/node"` is displayed, which doesn't convey useful information about the actual language server script being executed.

## Solution

There was a PR #53076 in which I was involved, and the solution there was to populate every language server adapter with a special marker for the path to be shown in the tooltip. That solution is accurate, but in order to make this solution work for extension-provided servers, the final diff became huge for a simple fix.

So here, as commented in https://github.com/zed-industries/zed/pull/53076#pullrequestreview-4204849892, a guess is performed by a newly introduced function `tooltip_for_server_binary()` to get the real path to be shown. It may have some edge cases, but works for current cases and is simple to implement.

## Testing

Added new unit tests, built and tested locally, with the comparision attached in the Showcase section.

## Self-Review Checklist:

- [x] I've reviewed my own diff for quality, security, and reliability
- [x] Unsafe blocks (if any) have justifying comments
- [x] The content adheres to Zed's UI standards ([UX/UI](https://github.com/zed-industries/zed/blob/main/CONTRIBUTING.md#uiux-checklist) and [icon](https://github.com/zed-industries/zed/blob/main/crates/icons/README.md) guidelines)
- [x] Tests cover the new/changed behavior
- [x] Performance impact has been considered and is acceptable

## Showcase

For the mentioned `Basedpyright` language server case, the comparision is shown below:
| Before | After |
|:--:|:--:|
| before | after |

---

Release Notes:

- Improved the LSP status tooltip to display the target script path for runtime-managed language servers

### CODE REVIEW

File: crates/language_tools/src/lsp_button.rs [hunk 1]
```diff
@@ -675,6 +679,34 @@ impl LanguageServerState {
...
+        .and_then(|_runtime| {
+            server_binary
+                .arguments
+                .iter()
+                .find(|arg| !arg.to_string_lossy().starts_with('-'))
+        })
+        .map(Path::new)
+        .unwrap_or(&server_binary.path);
+
+    let display_path = path_style.normalize(&target_path.compact().to_string_lossy());
```
Comment (lingyaochu) [inline crates/language_tools/src/lsp_button.rs:711]: We need a normalize here as some binary path is like "languages\\basedpyright\\node_modules/basedpyright/langserver.index.js" on windows, as covered in the test. We can also do some changes on language server adapter side case by case, by I don't think it's worthy.

File: crates/language_tools/src/lsp_button.rs [hunk 2]
```diff
@@ -260,24 +261,32 @@ impl LanguageServerState {
...
-                                status.binary.as_ref().map(|b| b.path.clone()),
-                                status.process_id,
-                            ),
-                        )
-                    })
-                    .collect::<HashMap<_, _>>()
-            })
-            .unwrap_or_default();
+                                server_id,
+                                (
```
Comment (MrSubidubi) [inline crates/language_tools/src/lsp_button.rs]: Not your fault at all, but since I have you here, can we make the server metadata a struct and derive default on that

File: crates/language_tools/src/lsp_button.rs [hunk 3]
```diff
@@ -675,6 +679,34 @@ impl LanguageServerState {
     }
 }
 
+fn tooltip_for_server_binary(
+    server_binary: &lsp::LanguageServerBinary,
+    path_style: PathStyle,
+) -> String {
```
Comment (MrSubidubi) [inline crates/language_tools/src/lsp_button.rs]: Should we make the return type a sharedString here already so that we know cloning that is cheap?

File: crates/language_tools/src/lsp_button.rs (hunk 1 above)
Comment (MrSubidubi) [reply crates/language_tools/src/lsp_button.rs:711]: Yep sounds fine with me too

File: crates/language_tools/Cargo.toml [hunk 4]
```diff
@@ -37,6 +37,7 @@ ui.workspace = true
 util.workspace = true
 workspace.workspace = true
 zed_actions.workspace = true
+path.workspace = true
```
Comment (MrSubidubi) [inline crates/language_tools/Cargo.toml]: Let's keep them sorted here, please

**Review by MrSubidubi — CHANGES_REQUESTED** [review]
Image

Few notes, but the fix itself looks really good and definitely better than the approach before

File: crates/language_tools/Cargo.toml (hunk 4 above)
Comment (lingyaochu) [reply crates/language_tools/Cargo.toml]: Done. Learned another thing to check before opening a PR.

File: crates/language_tools/src/lsp_button.rs (hunk 2 above)
Comment (lingyaochu) [reply crates/language_tools/src/lsp_button.rs]: Makes sense, definitely much cleaner. Updated!

File: crates/language_tools/src/lsp_button.rs (hunk 3 above)
Comment (lingyaochu) [reply crates/language_tools/src/lsp_button.rs]: Yeah, a better choice. Done.

**Review by MrSubidubi — APPROVED** [review]
<details>

Image

</details>

A pleasure as always, awesome to have you aroud. Thank you!
